src/Stripe.net/Infrastructure/Public/SystemNetHttpClient.cs

```diff
@@ -0,0 +1,129 @@
+namespace Stripe
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+    using Stripe.Infrastructure;
+
+    /// <summary>
+    /// Standard client to make requests to Stripe's API, using
+    /// <see cref="System.Net.Http.HttpClient"/> to send HTTP requests.
+    /// </summary>
+    public class SystemNetHttpClient : HttpClient
+    {
+        private static readonly string UserAgentString
+            = $"Stripe/v1 .NetBindings/{StripeConfiguration.StripeNetVersion}";
+
+        private static readonly string StripeClientUserAgentString
+            = BuildStripeClientUserAgentString();
+
+        private readonly System.Net.Http.HttpClient httpClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SystemNetHttpClient"/> class.
+        /// </summary>
+        /// <param name="httpClient">
+        /// The <see cref="System.Net.Http.HttpClient"/> client to use. If <c>null</c>, an HTTP
+        /// client will be created with default parameters.
+        /// </param>
+        public SystemNetHttpClient(System.Net.Http.HttpClient httpClient = null)
+        {
+            this.httpClient = httpClient ?? BuildDefaultSystemNetHttpClient();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="System.Net.Http.HttpClient"/> class
+        /// with default parameters.
+        /// </summary>
+        /// <returns>The new instance of the <see cref="System.Net.Http.HttpClient"/> class.</returns>
+        public static System.Net.Http.HttpClient BuildDefaultSystemNetHttpClient()
+        {
+            // We set the User-Agent and X-Stripe-Client-User-Agent headers in each request
+            // message rather than through the client's <c>DefaultRequestHeaders</c> because we
+            // want these headers to be present even when a custom HTTP client is used.
+            return new System.Net.Http.HttpClient()
+            {
+                Timeout = StripeConfiguration.DefaultHttpTimeout,
+            };
+        }
+
+        /// <summary>Sends a request to Stripe's API as an asynchronous operation.</summary>
+        /// <param name="request">The parameters of the request to send.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
+        /// <returns>The task object representing the asynchronous operation.</returns>
+        public override async Task<StripeResponse> MakeRequestAsync(
+            StripeRequest request,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var httpRequest = BuildRequestMessage(request);
+
+            this.LastRequest = request;
```

Comment:
While we're on the subject: should you synchronize access when setting these properties?

Based on my very [surface-level reading on the task model](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/task-based-asynchronous-programming), it looks like tasks may or may not run on different threads. Is the idea here that there could be a conflict on set, but it doesn't matter?

Reply:
I've removed the properties so it doesn't really matter anymore, but I think it wouldn't have mattered, but only because nothing relied on these properties. With multiple async requests and/or concurrent requests from different threads, you'd have no guarantee about the values.

In retrospect, removing them was probably the right call :) Thanks Brandur!